avocado/utils/process.py

```diff
@@ -99,10 +100,24 @@ def __str__(self):
             return "CmdError"
 
 
+def normalize_cmd(cmd):
+    """
+    Normalize cmd to be safe for ``shlex.split``
+    """
+    if PY2:
+        if not isinstance(cmd, str):
+            cmd = cmd.encode("utf-8")
+    else:
+        if isinstance(cmd, bytes):
```

Comment:
On the commit message you state that:

"The `shlex.split` does not support "unicode" strings on PY2"

But, this is what I get:

```
python2.7 -c "import shlex; print(shlex.split(u'foo bar baz'))"
['foo', 'bar', 'baz']
```

For Python 3, your statement seems correct:

```
$ python3 -c "import shlex; print(shlex.split(b'foo bar baz'))"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib64/python3.6/shlex.py", line 305, in split
    return list(lex)
  File "/usr/lib64/python3.6/shlex.py", line 295, in __next__
    token = self.get_token()
  File "/usr/lib64/python3.6/shlex.py", line 105, in get_token
    raw = self.read_token()
  File "/usr/lib64/python3.6/shlex.py", line 136, in read_token
    nextchar = self.instream.read(1)
AttributeError: 'bytes' object has no attribute 'read'
```

But I wonder why we should attempt to do the conversion.  It seems to be an incorrect usage.

Reply:
In py2.7.3 they added support for unicode in ascii range (actually ISO something), but if you provide some unicode value it will fail.

On py3 string is actually unicode and shlex works well with it. On the other hand py3 encodes into bytes, but bytes are not accessible by bytes and shlex fails on it. This is why there is the `if PY2` and encoding in py2 or decoding on py3.

Reply:
OK, now I see what you mean.  It's not about the string "type" (that is, unicode or str).  If we provide an non-ascii code point, `shlex.split()` will fail:

```
python2.7 -c "import shlex; print(shlex.split(u'\xaefoo bar baz'))"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib64/python2.7/shlex.py", line 275, in split
    lex = shlex(s, posix=posix)
  File "/usr/lib64/python2.7/shlex.py", line 25, in __init__
    instream = StringIO(instream)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xae' in position 0: ordinal not in range(128)
```